operator/v1/types_network.go

```diff
@@ -256,6 +256,40 @@ type KuryrConfig struct {
 	// size by 1.
 	// +optional
 	OpenStackServiceNetwork string `json:"openStackServiceNetwork,omitempty"`
+
+	// enableDebug set to true will enable debug logs in kuryr-kuberntes.
+	// +optional
+	EnableDebug *bool `json:"enableDebug,omitempty"`
+
+	// enablePortPrepopulation when true will make Kuryr prepopulate each newly created port
+	// pool with a minimum number of ports. Kuryr uses Neutron port pooling to fight the fact
+	// that it takes a significant amount of time to create one. Instead of creating it when
+	// pod is being deployed, Kuryr keeps a number of ports ready to be attached to pods.
+	// +optional
+	EnablePortPrepopulation *bool `json:"enablePortPrepopulation,omitempty"`
+
+	// poolMaxPorts sets a maximum number of free ports that are being kept in a port pool.
+	// If the number of ports exceeds this setting, free ports will get deleted. Setting 0
+	// will disable this. For more information about port pools see enablePortPrepopulation
+	// setting.
+	// +kubebuilder:validation:Minimum=0
+	// +optional
+	PoolMaxPorts *int `json:"poolMaxPorts,omitempty"`
```

Comment:
why do I have this setting *and* `enablePortPrepoulation`.  Does it make sense for enablePortPrepopulation to be true and this to be zero?  Does it make sense for this to be non-zero and enablePortPrepopulation to be false?

Reply:
Right, so there are two levels of this feature that can be enabled in Kuryr:

- Port pools - with OpenShift 4.2 this is enabled and we don't give users ability to disable that.
- If port pools are enabled, you can enable pools prepopulation. When enabled each pool will get ports created when a new namespace gets created. This defaults to false in OpenShift, but if you don't create a lot of namespaces on your env it might have sense to enable it and speed things up a bit.

All this comes from the fact that OpenStack Neutron is really slow when creating ports.

It makes total sense to have `enablePortPrepopulation=true` and `PoolMaxPorts=0`, the latter is just upper bound. Also `enablePortPrepopulation=false` and `PoolMaxPorts=x` make sense, those are separate.

I guess i can rename `enablePortPrepopulation` to `enablePortPoolsPrepopulation`?